lib/util/package-rules/package-names.ts

```diff
@@ -14,15 +15,16 @@ export class PackageNameMatcher extends Matcher {
       return false;
     }
 
-    if (matchPackageNames.includes(depName)) {
+    if (matchPackageNames.includes(packageName || depName)) {
```

Comment:
```suggestion
    if (matchPackageNames.includes(packageName)) {
```
otherwise you never get into condition below?

Reply:
Actually it's intentional. We only want to warn if packageName and depName are separate, AND it only matches depName. The way I structured it was intended to achieve that, can you take another look with that perspective?

Reply:
this condition only returns true if packageName is null, undefined, empty string, 0, false and depName is in the list. so this check looks strange to me as package name is required (isn't it?) so if package name is matching, we return true here and the check below never gets evaluated. 🤔

Reply:
What if packageName != depName? That's the only case we care about warning about

Reply:
Check the second added test

Reply:
@viceice please see above

Reply:
still looks strange to me but ok i would suggest this change

```suggestion
    if (matchPackageNames.includes(packageName ?? depName)) {
```